File: Customer App/lib/models/banner.dart
class PromoBanner {
  final String id;
  final String title;
  final String imageUrl;
  final String targetScope;
  final String? targetCity;
  final String? targetSalonId;
  final String? actionUrl;

  PromoBanner({
    required this.id,
    required this.title,
    required this.imageUrl,
    required this.targetScope,
    this.targetCity,
    this.targetSalonId,
    this.actionUrl,
  });

  factory PromoBanner.fromJson(Map<String, dynamic> json) {
    return PromoBanner(
      id: json['id'],
      title: json['title'],
      imageUrl: json['image_url'],
      targetScope: json['target_scope'],
      targetCity: json['target_city'],
      targetSalonId: json['target_salon_id'],
      actionUrl: json['action_url'],
    );
  }
}

File: backend/app/Models/Banner.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Concerns\HasUuids;

class Banner extends Model
{
    use HasUuids;

    public $timestamps = false;

    protected $fillable = [
        'title',
        'image_url',
        'action_url',
        'target_scope',
        'target_city_id',
        'target_salon_id',
        'start_date',
        'end_date',
        'is_active',
    ];
    
    protected $casts = [
        'is_active' => 'boolean',
        'start_date' => 'date',
        'end_date' => 'date',
    ];
}
